### ISSUE TITLE
License - Redeem confirmation

### ISSUE THREAD

**Issue by drazenbebic** [drazenbebic/license-manager-for-woocommerce#111]
Instead of sending the licenses via email, there should also be an option to send a "redeem" link which will lead them to a custom page on the website, where they have to confirm that they wish to redeem the license key. This should also be logged in the backend, so that the administrators know which licenses have been redeemed and which haven't.

https://wordpress.org/support/topic/confirmation-of-license-retrieval-by-customer/#post-12762422

**Comment by swspcodes** [comment]
It´s a good idea to integrate this "confirm" feature to the same area where the license codes are attached to the order page (and email), so that the current workflow for the shops can remain the same, but with the confirmation action added. 

Adding a new step/page for the license retreival may be very good, but it can also add more "difficulty" for the customers and for the shops, who will have to adapt to the new step added to the sales process.

Maybe two options: "Get confirmation in a custom page", and "Get confirmation on the order´s page" could be an idea for the backend settings?

**Comment by drazenbebic** [comment]
@swspcodes 

Absolutely! This will be introduced as an optional feature. I'm thinking that, by enabling this in the settings, instead of receiving your license keys immediately in the email, you will instead be given a link to the website to redeem your license. If the setting is off, the license keys will be included in the email as usual.

Another important thing is security. We do not want just anyone going to the link and redeeming license keys, so perhaps a password check will be necessary. Perhaps generating a unique hash which will be checked also makes sense. I'll see what can be done about this part.

**Comment by swspcodes** [comment]
Amazing!

You have a much greater view of how this can be done than I could ever
think of.

Can't wait for it.

I sell license codes from many manufacturers, not my own generated codes,
and my providers don't take returns once I request the keys... so it will
be very important to have this feature to protect us from chargebacks and
other types of problems

Thanks

Em ter, 5 de mai de 2020 03:51, Drazen Bebic <notifications@github.com>
escreveu:
